lib/plug/router.ex

```diff
@@ -8,18 +8,25 @@ defmodule Plug.Router do
         use Plug.Router
 
         plug :match
-        plug :dispatch
+        plug :dispatch, builder_opts()
+        
+        def init(_) do
+          [hello: "world"]
+        end
 
         get "/hello" do
-          send_resp(conn, 200, "world")
+          send_resp(conn, 200, opts[:content])
         end
 
         match _ do
           send_resp(conn, 404, "oops")
         end
       end
 
-  Each route needs to return a connection, as per the Plug spec.
+  Each route receives 2 variables `conn` of type `Plug.Conn.t()`
+  and `opts` of type `Plug.opts()` and needs to return a connection,
+  as per the Plug spec. The `opts` variables is assigned to the one
+  passed to `plug :dispatch`, see `Plug.Builder.builder_opts/0` for more details.
```

Comment:
I think using `Plug.Builder.builder_opts/0` in this example might be confusing at first. 

What do you think about moving this documentation to a different section (or maybe to the "Routes compilation" one?) That documentation would describe that you can pass options to the `:dispatch` function plug and that it also works with `Plug.Builder.builder_opts/0`.